email-smtp/it/ConfigurationIntegrationTest.java

```diff
@@ -0,0 +1,107 @@
+        try(ServerSocket serverSocket = new ServerSocket(openPortOnWhichNoConnectionsAreAccepted)) {
+            final SmtpAuthEmailSender sender = new SmtpAuthEmailSender(executor,
+                    setup.getBindAddress(), openPortOnWhichNoConnectionsAreAccepted,
+                    SmtpAuthEmailSender.TransportSecurity.None, USERNAME, PASSWORD, timeout300Ms);
+            assertThatThrownBy(() -> { sender.send(TestUtils.validShortEmail()).toCompletableFuture().join(); })
+                    .hasCauseInstanceOf(EmailSenderException.class)
+                    .hasStackTraceContaining("java.net.SocketTimeoutException: Read timed out");
+        }
+    }
+
+    @Test
+    public void portThatDoesNotSendGreetingRaisesException() throws Exception {
+        final int timeout300Ms = 300;
+        final int openPort = 12346;
+        try(ServerSocket serverSocket = new ServerSocket(openPort)) {
+            final Thread acceptor = new Thread(accept1ConnectionButDontWriteGreeting(serverSocket));
+            acceptor.start();
+            final SmtpAuthEmailSender sender = new SmtpAuthEmailSender(executor,
+                    setup.getBindAddress(), openPort,
+                    SmtpAuthEmailSender.TransportSecurity.None, USERNAME, PASSWORD, timeout300Ms);
+            assertThatThrownBy(() -> { sender.send(TestUtils.validShortEmail()).toCompletableFuture().join(); })
+                    .hasCauseInstanceOf(EmailSenderException.class)
+                    .hasStackTraceContaining("java.net.SocketTimeoutException: Read timed out");
+        }
+    }
+
+    private IORunnable accept1ConnectionButDontWriteGreeting(@Nonnull final ServerSocket serverSocket) {
+        return () -> { serverSocket.accept(); };
+    }
+
+    // Note: it would be desireable to test the write timeout of the Java Mail API, but
+    // I cannot think of an easy way to make it write enough data that causes the TCP stack
+    // to pause writes, because the Java Mail API likely waits for server acknowledgements
+    // before it continues sending data.
+
+    @Test
+    public void twoSendersForDifferentPortsDoNotInterfereWithEachOther() throws Exception {
+        final int invalidPort = 123;
+        final SmtpAuthEmailSender dysfunctionalSender = new SmtpAuthEmailSender(executor,
+                setup.getBindAddress(), invalidPort,
+                SmtpAuthEmailSender.TransportSecurity.None, USERNAME, PASSWORD, TIMEOUT_60_SECONDS);
+
+        String messageId = sender.send(TestUtils.validShortEmail()).toCompletableFuture().join();
+
+        assertThat(messageId).isNotEmpty();
+    }
+
+
+    /**
+     * Used to re-throw {@link IOException}s as unchecked exceptions.
+     */
+    private interface IORunnable extends Runnable {
+
+        void io() throws IOException;
+
+        @Override
+        default void run() {
+            try {
+                io();
+            } catch (IOException e) {
+                throw new IllegalStateException("I/O failed", e);
```

Comment:
Do not change this here.

Normally https://docs.oracle.com/javase/8/docs/api/java/io/UncheckedIOException.html can be used to get an unchecked IOException.